packages/ui-kit/src/components/AccessTokenProvider/AccessTokenProvider.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { createContext, PropsWithChildren, useCallback, useEffect, useRef, useState } from 'react'
+import { useLog } from '../Log'
+
+interface AccessTokenContextValue {
+  accessToken?: string
+  isLoading?: boolean
+  onExpiredToken?: () => void
+  failedRetry?: boolean
+}
```

Comment:
Can you document each of these? For example, I am wondering what the point of propagating `failedRetry` is.